ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -245,24 +261,109 @@ static int
-    WT_UNUSED(file_system);
     WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(file_type);
-    WT_UNUSED(flags);
-    WT_UNUSED(file_handle_ptr);
+
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    bool exists;
+    size_t size;
+
+    *file_handle_ptr = nullptr;
+
+    // We only support opening the file in read only mode.
+    if ((flags & WT_FS_OPEN_READONLY) == 0 || (flags & WT_FS_OPEN_CREATE) != 0) {
+        std::cerr << "gcp_file_open: read-only access required." << std::endl;
+        return EINVAL;
+    }
+
+    // Currently, only data files should be being opened; although this constraint can be relaxed in
+    // the future.
+    if (file_type != WT_FS_OPEN_FILE_TYPE_DATA && file_type != WT_FS_OPEN_FILE_TYPE_REGULAR) {
+        std::cerr << "gcp_file_open: only data file and regular types supported." << std::endl;
+        return EINVAL;
+    }
+
+    // Check if object exists in the cloud.
+    if ((fs->gcp_conn->object_exists(name, exists, size) == 0) && !exists) {
+        std::cerr << "gcp_file_open: object with name " << name << " does not exist in the bucket."
+                  << std::endl;
+        return EINVAL;
+    }
+
+    // Check if there is already an existing file handle open.
+    std::vector<gcp_file_handle *>::iterator fh_iterator =
+      std::find_if(fs->fh_list.begin(), fs->fh_list.end(),
+        [name](gcp_file_handle *fh) { return strcmp(name, fh->name.c_str()) == 0; });
+
+    if (fh_iterator != fs->fh_list.end()) {
+        (*fh_iterator)->reference_count++;
+        *file_handle_ptr = reinterpret_cast<WT_FILE_HANDLE *>(*fh_iterator);
+
+        return 0;
+    }
+
+    // If file handle does not exist.
+    gcp_file_handle *gcp_fh;
+    try {
+        gcp_fh = new gcp_file_handle;
+    } catch (std::bad_alloc &e) {
+        std::cerr << "gcp_file_open: " << e.what() << std::endl;
+        return ENOMEM;
+    }
+    gcp_fh->file_system = fs;
+    gcp_fh->name = std::string(name);
+    gcp_fh->reference_count = 1;
+
+    WT_FILE_HANDLE *file_handle = &gcp_fh->fh;
+    gcp_fh->fh.close = gcp_file_close;
+    gcp_fh->fh.fh_advise = nullptr;
+    gcp_fh->fh.fh_extend = nullptr;
+    gcp_fh->fh.fh_extend_nolock = nullptr;
+    gcp_fh->fh.fh_lock = gcp_file_lock;
+    gcp_fh->fh.fh_map = nullptr;
+    gcp_fh->fh.fh_map_discard = nullptr;
+    gcp_fh->fh.fh_map_preload = nullptr;
+    gcp_fh->fh.fh_unmap = nullptr;
+    gcp_fh->fh.fh_read = gcp_file_read;
+    gcp_fh->fh.fh_size = gcp_file_size;
+    gcp_fh->fh.fh_sync = nullptr;
+    gcp_fh->fh.fh_sync_nowait = nullptr;
+    gcp_fh->fh.fh_truncate = nullptr;
+    gcp_fh->fh.fh_write = nullptr;
+
+    // We require exclusive access to the list of file handles when adding file handles to it. The
+    // lock_guard will be unlocked automatically when the scope is exited.
+    {
+        std::lock_guard<std::mutex> lock(fs->fh_list_mutex);
+        fs->fh_list.push_back(gcp_fh);
+    }
+
+    *file_handle_ptr = file_handle;
```

Comment:
We don't need file_handle variable here, since we can replace it with gcp_fh->fh.

Reply:
addressed here:[687beb0](https://github.com/wiredtiger/wiredtiger/pull/8777/commits/687beb0e754c4e60654dfe49bdeda9dc9bf03665)